### ISSUE TITLE
etag.replace is not a function

### ISSUE THREAD

**Issue by cluzier** [perry-mitchell/webdav-client#162]
Error: `etag.replace is not a function`

Here is my code:
```
const { createClient } = require("webdav");

const client = createClient(
    "http://localhost:4200",
    {
        username: "username",
        password: "password"
    }
);
async function fileupdate() {
    // Get directory contents
    try {
        const directoryItems = await client.getDirectoryContents("/");
        console.log(directoryItems);
    }
    catch (ex) { 
        console.log(ex.message) 
    }
};

fileupdate();
```

**Comment by perry-mitchell** [comment]
The code that calls that is [here](https://github.com/perry-mitchell/webdav-client/blob/d8cd0c65a021f9361e7b3886adb25a9a3607c648/source/interface/dav.js#L60), but I can’t see how it’d fail unless the etag wasn’t a string. Any way you could maybe drop a console.log in that code while running yours so we can see what the etag is?

Of course I understand if you’ve moved on from this.

**Comment by perry-mitchell** [comment]
Another idea would be to test that the etag is a string rather than assuming it is right now - I’ll mark this as a bug for that reason.

**Comment by perry-mitchell** [comment]
Released a fix for this in 2.9.1, which now checks that it is a string.
